Content.Server/Anomaly/Effects/ReagentProducerAnomalySystem.cs

```diff
@@ -0,0 +1,199 @@
+            component.RealReagentProducing *= component.SupecriticalReagentProducingModifier;
+    }
+
+    /// <summary>
+    /// returns a random reagent based on a system of random weights.
+    /// First, the category is selected: The category has a minimum and maximum weight,
+    /// the current value depends on severity.
+    /// Accordingly, with the strengthening of the anomaly,
+    /// the chances of falling out of some categories grow, and some fall.
+    ///
+    /// After that, a random reagent in the selected category is selected.
+    ///
+    /// Such a system is made to control the danger and interest of the anomaly more.
+    ///
+    /// ATTENTION! All new reagents appearing in the game will automatically appear in the "Other" category
+    /// </summary>
+    private string GetRandomReagentType(EntityUid uid, ReagentProducerAnomalyComponent component, float severity)
+    {
+        //Category Weight Randomization
+        var currentWeightDangerous = Lerp(component.WeightSpreadDangerous.X, component.WeightSpreadDangerous.Y, severity);
+        var currentWeightFun = Lerp(component.WeightSpreadFun.X, component.WeightSpreadFun.Y, severity);
+        var currentWeightUseful = Lerp(component.WeightSpreadUseful.X, component.WeightSpreadUseful.Y, severity);
+        var currentWeightOther = Lerp(component.WeightSpreadOther.X, component.WeightSpreadOther.Y, severity);
+
+        var sumWeight = currentWeightDangerous + currentWeightFun + currentWeightUseful + currentWeightOther;
+        var rnd = _random.NextFloat(0f, sumWeight);
+
+
+        if (rnd <= currentWeightDangerous && component.DangerousChemicals.Count > 0)
+        {
+            var reagent = _random.Pick(component.DangerousChemicals);
+            return reagent;
+        }
+        else rnd -= currentWeightDangerous;
+
+        if (rnd <= currentWeightFun && component.FunChemicals.Count > 0)
+        {
+            var reagent = _random.Pick(component.FunChemicals);
+            return reagent;
+        }
+        else rnd -= currentWeightFun;
+
+        if (rnd <= currentWeightUseful && component.UsefulChemicals.Count > 0)
+        {
+            var reagent = _random.Pick(component.UsefulChemicals);
+            return reagent;
+        }
+        else //Pickup other
+        {
+            var allReagents = _proto.EnumeratePrototypes<ReagentPrototype>().Select(proto => proto.ID).ToHashSet();
+
+            foreach (var chem in component.BlacklistChemicals)
+            {
+                allReagents.Remove(chem);
+            }
+            foreach (var chem in component.DangerousChemicals)
+            {
+                allReagents.Remove(chem);
+            }
+            foreach (var chem in component.FunChemicals)
+            {
+                allReagents.Remove(chem);
+            }
+            foreach (var chem in component.UsefulChemicals)
+            {
+                allReagents.Remove(chem);
+            }
```

Comment:
yea, i really dislike this whitelist + blacklist scheme, creates a lot of jank, i would rather just have the whitelists